advocacy_docs/edb-postgres-ai/migration-etl/data-migration-service/getting_started/prepare_schema.mdx

```diff
@@ -0,0 +1,62 @@
+
+EDB recommends [applying the following constraints](apply_constraints) on the target database after you have signalized the end of the CDC stream by [marking the migration as completed](mark_completed) in the Console.
+
+`FOREIGN KEY` / `REFERENCES`
+
+`CHECK`
+
+`CASCADE`
+
+`EXCLUDE` 
+
+## Preparing your schema
+
+You can use several different methods to prepare your schema, exclude unsupported constraints and import it to the target database. This section provides guidelines of how to prepare your Oracle or Postgres source database with Migration Portal or Migration Toolkit, but other tools like `pg_dump` and `pg_restore` are also valid routes for migrating DDL.
+
+### Oracle to EDB Postgres Advanced Server migrations
+
+Use [EDB Migration Portal](/migration_portal/latest/03_mp_using_portal/03_mp_quick_start/) to assess Oracle database sources for schema compatibility before starting the data migration process. 
+
+EDB Migration Portal offers the ability to separate constraints from other destination DDL with the [offline migration option](/migration_portal/latest/04_mp_migrating_database/03_mp_schema_migration/#offline-migration).
+
+Ensure you exclude `FOREIGN KEY`, `REFERENCES`, `CHECK`, `CASCADE` and `EXCLUDE` constraints from the DDL before importing the schema to the target database.
+
+### Other migrations
+
+For data migrations from Postgres, EDB recommends using [EDB Migration Toolkit](/migration_toolkit/latest/) to manage the schema. MTK's [offline migration](/migration_toolkit/latest/07_invoking_mtk/08_mtk_command_options/#offline-migration-options) capability provides an easy way to extract a database's schema and separate constraints.
+
+Ensure you exclude `FOREIGN KEY`, `REFERENCES`, `CHECK`, `CASCADE` and `EXCLUDE` constraints from the DDL before importing the schema to the target database.
+
+## Importing your schema to the target database
+
+After you have prepared the DDL, and excluded `FOREIGN KEY`, `REFERENCES`, `CHECK`, `CASCADE` and `EXCLUDE` constraints, connect to the target database and import the SQL-formatted DDL file.
+
+Importing the schema to the target database before the migration takes place is important as EDB DMS only migrates data. The target database must have the schemas in place for the migration to populate them with data.
+
+You can use different methods to import the schemas. You can manually create them in the target database, you can perform a [pg_restore](https://www.postgresql.org/docs/current/app-pgrestore.html) (for example, if you used [pg_dump](https://www.postgresql.org/docs/current/app-pgdump.html) to obtain the schemas), you can employ [pgAdmin](https://www.pgadmin.org/docs/pgadmin4/latest/index.html), [psql](https://www.postgresql.org/docs/7.0/app-psql.htm) or the tool of your preference.
```

Comment:
Agree, this is a bit difficult to phrase, because it is important to separate the import from export, so the customer can manually discard the unsupported constraints as an in-between step. MP and MTK can do this a different levels, depending on the mode they use (online/offline). Let me try to rephrase this according to what you suggested and what I know based on the docs. Let me know what you think: 

```suggestion
You can use different methods to import the schemas.

If you used Migration Portal to export an Oracle source schema, continue using it to [import the schema in offline or online mode](/migration_portal/latest/04_mp_migrating_database/03_mp_schema_migration/#mp_schema_migration).

If you used Migration Toolkit to export a Postgres source in offline mode, execute the generated [offline migration script to start the import](/migration_toolkit/latest/07_invoking_mtk/08_mtk_command_options/#offline-migration-options). 

Other alternatives to import Postgres schemas include manually creating the schemas in the target database, performing a [pg_restore](https://www.postgresql.org/docs/current/app-pgrestore.html) (for example, if you used [pg_dump](https://www.postgresql.org/docs/current/app-pgdump.html) to obtain the schemas), or employing [pgAdmin](https://www.pgadmin.org/docs/pgadmin4/latest/index.html), [psql](https://www.postgresql.org/docs/7.0/app-psql.htm) or the tool of your preference.
```

Reply:
I think your proposed change is better.